src/Symfony/Component/Validator/Resources/bin/sync-iban-formats.php

```diff
@@ -0,0 +1,204 @@
+
+            foreach ($this->parseCountryCodesList($item['included_country_codes']) as $includedCountryCode) {
+                $formats[$includedCountryCode] = $formats[$item['country_code']];
+            }
+        }
+
+        return $formats;
+    }
+
+    /**
+     * @return list<string>
+     */
+    private function parseCountryCodesList(string $countryCodesList): array
+    {
+        if ('N/A' === $countryCodesList) {
+            return [];
+        }
+
+        $countryCodes = [];
+
+        foreach (explode(',', $countryCodesList) as $countryCode) {
+            $countryCodes[] = preg_replace('/^([A-Z]{2})(\s+\(.+?\))?$/', '$1', trim($countryCode));
+        }
+
+        return $countryCodes;
+    }
+
+    /**
+     * @param array<string, string> $properties
+     *
+     * @return list<array<string, string>>
+     */
+    private function readPropertiesFromRegistry(array $properties): array
+    {
+        $items = [];
+
+        $registryContent = file_get_contents('https://www.swift.com/swift-resource/11971/download');
+        $lines = explode("\n", $registryContent);
+
+        // skip header line
+        array_shift($lines);
+
+        foreach ($lines as $line) {
+            $columns = str_getcsv($line, "\t");
+            $propertyLabel = array_shift($columns);
+
+            if (!isset($properties[$propertyLabel])) {
+                continue;
+            }
+
+            $propertyField = $properties[$propertyLabel];
+
+            foreach ($columns as $index => $value) {
+                $items[$index][$propertyField] = $value;
+            }
+        }
+
+        return array_values($items);
+    }
+
+    private function buildIbanRegexp(string $ibanStructure): string
+    {
+        $pattern = $ibanStructure;
+
+        $pattern = preg_replace('/(\d+)!n/', '\\d{$1}', $pattern);
+        $pattern = preg_replace('/(\d+)!a/', '[A-Z]{$1}', $pattern);
+        $pattern = preg_replace('/(\d+)!c/', '[\\dA-Z]{$1}', $pattern);
+
+        return $pattern;
+    }
+}
+
+final class WikipediaIbanProvider
+{
+    /**
+     * @return array<string, array<string, string>>
+     */
+    public function getIbanFormats(): array
+    {
+        $formats = [];
+
+        foreach ($this->readIbanFormatsTable() as $item) {
+            if (!preg_match('/^([A-Z]{2})/', $item['Example'], $matches)) {
+                continue;
+            }
+
+            $countryCode = $matches[1];
+
+            $formats[$countryCode] = [$this->buildIbanRegexp($countryCode, $item['BBAN Format']), $item['Country']];
+        }
+
+        return $formats;
+    }
+
+    /**
+     * @return list<array<string, string|int>>
+     */
+    private function readIbanFormatsTable(): array
+    {
+        $tablesResponse = file_get_contents('https://www.wikitable2json.com/api/International_Bank_Account_Number?table=3&keyRows=1&clearRef=true');
+
+        return json_decode($tablesResponse, true)[0];
```

Comment:
I suggest using `JSON_ERROR_THROW` here, to get a proper failure in case we receive invalid JSON. It would make the script more robust.

Reply:
fixed thanks